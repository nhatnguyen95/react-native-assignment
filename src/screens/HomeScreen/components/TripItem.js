import React from "react";
import PropTypes from "prop-types";
import { TouchableOpacity, Text, StyleSheet } from "react-native";

const TripItem = ({ id = 0, onPress = () => {} }) => (
  <TouchableOpacity style={styles.container} onPress={onPress}>
    <Text style={styles.tripIdText}>Trip Id: {id}</Text>
  </TouchableOpacity>
);

TripItem.propTypes = {
  id: PropTypes.number,
  onPress: PropTypes.func,
};


const styles = StyleSheet.create({
  container: {
    justifyContent: 'center',
    alignItems: 'center',
    paddingVertical: 20,
    backgroundColor: '#fff',
    marginTop: 1,
  },
  tripIdText: {
    fontSize: 16,
  }
})

export default TripItem;
