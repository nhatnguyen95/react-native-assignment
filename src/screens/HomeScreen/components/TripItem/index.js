import React from "react";
import PropTypes from "prop-types";
import { TouchableOpacity, Text, StyleSheet } from "react-native";
import Strings from "constants/Strings";

const TripItem = ({ id = 0, onPress = () => {} }) => (
  <TouchableOpacity style={styles.container} onPress={onPress}>
    <Text style={styles.tripIdText}>{Strings.TRIP_ID}: {id}</Text>
  </TouchableOpacity>
);

TripItem.propTypes = {
  id: PropTypes.number,
  onPress: PropTypes.func,
};

const styles = StyleSheet.create({
  container: {
    justifyContent: "center",
    alignItems: "center",
    paddingVertical: 20,
    backgroundColor: "#fff",
    marginTop: 1,
  },
  tripIdText: {
    fontSize: 16,
  },
});

export default TripItem;
