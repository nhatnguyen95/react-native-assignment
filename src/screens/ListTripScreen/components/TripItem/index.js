import React from "react";
import PropTypes from "prop-types";
import { TouchableOpacity, Text, StyleSheet } from "react-native";
import Strings from "constants/Strings";


const TripItem = ({ id = 0, onPress = () => {} }) => (
  <TouchableOpacity activeOpacity={0.8} style={styles.container} onPress={onPress}>
    <Text style={styles.tripIdText}>{Strings.TRIP_ID}: {id}</Text>
  </TouchableOpacity>
);

TripItem.propTypes = {
  id: PropTypes.number,
  onPress: PropTypes.func,
};

const styles = StyleSheet.create({
  container: {
    justifyContent: "center",
    alignItems: "center",
    paddingVertical: 20,
    borderRadius: 10,
    marginTop: 10,
    marginHorizontal: 10,
    backgroundColor: '#fff',
    borderWidth: 1,
    borderColor: '#51bd8a',
    paddingHorizontal: 10,
  },
  tripIdText: {
    fontSize: 19,
    color: '#51bd8a',
    fontWeight: 'bold'
  },
});

export default TripItem;
