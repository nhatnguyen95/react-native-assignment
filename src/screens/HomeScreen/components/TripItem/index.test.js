import "react-native";
import React from "react";

import TripItem from './index';

import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import { Text } from "react-native";

describe("TripItem", () => {
  describe("Rendering", () => {
    it("should match to snapshot", () => {
      const wrapper = renderer.create(<TripItem />).toJSON();
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe("press on TripItem", () => {
    it("should call onPress", () => {
      const mockOnPress = jest.fn();
      const wrapper = shallow(<TripItem onPress={mockOnPress} />);
      const onPress = wrapper.prop("onPress");
      onPress();
      expect(mockOnPress).toHaveBeenCalled();
    });
  });
  
  describe("pass id to trip", () => {
    it("should equal 10", () => {
      const wrapper = shallow(<TripItem id={10} />);
      const text = wrapper.find(Text);
      const id = text.props().children[2];
      expect(id).toBe(10);
    });
  });
  
});

