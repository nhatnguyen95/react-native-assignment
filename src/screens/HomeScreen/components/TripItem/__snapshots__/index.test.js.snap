// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TripItem Rendering should match to snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "justifyContent": "center",
      "marginTop": 1,
      "opacity": 1,
      "paddingVertical": 20,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 16,
      }
    }
  >
    Trip Id
    : 
    0
  </Text>
</View>
`;
